dojo/pipeline.py

```diff
@@ -183,5 +183,10 @@ def sanitize_username(username):
 def create_user(strategy, details, backend, user=None, *args, **kwargs):
     if not settings.SOCIAL_AUTH_CREATE_USER:
         return None
-    details["username"] = sanitize_username(details.get("username"))
+    username = details.get("username")
+    if not username:
+        username = details.get("email")
+    if not username:
+        username = details.get("fullname")
+    details["username"] = sanitize_username(username)
```

Comment:
I don't know if it's the best idea. If `username` is missing, it might mean that _social_ is not configured well. This might:
- point to unexpected behaviour (you do not expect by default that some other fields is used as username)
- create a security hole: if changing of email or fullname is fully in the hands of the user (in IDP), the user might impersonate another user.

I'm not against this change in general. It just should not be on by default.